File: frontend/src/app/auth/jwt-response.ts
export class JwtResponse {
    token: string;
    type: string;
    nickname: string;
    roles: string[];

    constructor(token: string, type: string, nickname: string, roles: string) {
        this.token = token;
        this.type = type;
        this.nickname = nickname;
        this.roles = ['user'];
    }
}

File: src/main/java/com/doodle/authorization/pojo/JwtResponse.java
package com.doodle.authorization.pojo;

import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.UUID;

@Getter
@Setter
public class JwtResponse {

    private String token;
    private String type = "Bearer";
    private UUID id;
    private String nickname;
    private String email;
    private List<String> roles;

    public JwtResponse(String token, UUID id, String nickname, String email, List<String> roles) {
        this.token = token;
        this.id = id;
        this.nickname = nickname;
        this.email = email;
        this.roles = roles;
    }
}
